core/drivers/firewall/stm32_risaf.c

```diff
@@ -0,0 +1,770 @@
+		risaf_sec = ((cfg & _RISAF_REG_CFGR_SEC) >>
+			     _RISAF_REG_CFGR_SEC_SHIFT) << DT_RISAF_SEC_SHIFT;
+		risaf_enc = ((cfg & _RISAF_REG_CFGR_ENC) >>
+			     _RISAF_REG_CFGR_ENC_SHIFT) <<
+			    (DT_RISAF_ENC_SHIFT + 1);
+		risaf_priv = ((cfg & _RISAF_REG_CFGR_PRIVC_MASK) >>
+			      _RISAF_REG_CFGR_PRIVC_SHIFT) <<
+			     DT_RISAF_PRIV_SHIFT;
+		cid_cfg = io_read32(base + _RISAF_REG_CIDCFGR(id));
+		rden = (cid_cfg & _RISAF_REG_CIDCFGR_RDENC_MASK) <<
+		       DT_RISAF_READ_SHIFT;
+		wren = ((cid_cfg & _RISAF_REG_CIDCFGR_WRENC_MASK) >>
+			_RISAF_REG_CIDCFGR_WRENC_SHIFT) << DT_RISAF_WRITE_SHIFT;
+		risaf->pdata.regions[i].cfg = id | risaf_en | risaf_sec |
+					      risaf_enc | risaf_priv |
+					      rden | wren;
+		start_addr = io_read32(base + _RISAF_REG_STARTR(id));
+		end_addr = io_read32(base + _RISAF_REG_ENDR(id));
+		risaf->pdata.regions[i].addr = start_addr +
+					       risaf->pdata.mem_base;
+		risaf->pdata.regions[i].len = end_addr - start_addr + 1U;
+	}
+
+	return TEE_SUCCESS;
+}
+
+static TEE_Result
+stm32_risaf_pm(enum pm_op op, unsigned int pm_hint __unused,
+	       const struct pm_callback_handle *pm_handle)
+{
+	struct stm32_risaf_instance *risaf = pm_handle->handle;
+	TEE_Result res = TEE_ERROR_GENERIC;
+
+	if (!PM_HINT_IS_STATE(pm_hint, CONTEXT))
+		return TEE_SUCCESS;
+
+	res = clk_enable(risaf->pdata.clock);
+	if (res)
+		return res;
+
+	if (op == PM_OP_RESUME)
+		res = stm32_risaf_pm_resume(risaf);
+	else
+		res = stm32_risaf_pm_suspend(risaf);
+
+	clk_disable(risaf->pdata.clock);
+
+	return res;
+}
+
+static TEE_Result stm32_risaf_probe(const void *fdt, int node,
+				    const void *compat_data)
+{
+	const struct stm32_risaf_compat_data *compat = compat_data;
+	struct stm32_risaf_instance *risaf = NULL;
+	struct stm32_risaf_region *regions = NULL;
+	TEE_Result res = TEE_ERROR_GENERIC;
+	struct dt_node_info dt_info = { };
+	const fdt32_t *conf_list = NULL;
+	const fdt64_t *cuint = NULL;
+	unsigned int nregions = 0;
+	unsigned int i = 0;
+	int len = 0;
+
+	res = stm32_rifsc_check_tdcid(&is_tdcid);
+	if (res)
+		return res;
+
+	if (!is_tdcid)
+		return TEE_SUCCESS;
+
+	risaf = calloc(1, sizeof(*risaf));
+	if (!risaf)
+		return TEE_ERROR_OUT_OF_MEMORY;
+
+	fdt_fill_device_info(fdt, &dt_info, node);
+	if (dt_info.reg == DT_INFO_INVALID_REG ||
+	    dt_info.reg_size == DT_INFO_INVALID_REG_SIZE) {
+		free(risaf);
+		return TEE_ERROR_BAD_PARAMETERS;
+	}
+
+	risaf->pdata.base.pa = dt_info.reg;
+	io_pa_or_va_secure(&risaf->pdata.base, dt_info.reg_size);
+
+	risaf->pdata.enc_supported = compat->supported_encryption;
+
+	res = clk_dt_get_by_index(fdt, node, 0, &risaf->pdata.clock);
+	if (!risaf->pdata.clock)
+		goto err;
+
+	conf_list = fdt_getprop(fdt, node, "memory-region", &len);
+	if (!conf_list) {
+		DMSG("RISAF %#"PRIxPTR": No configuration in DT, use default",
+		     risaf->pdata.base.pa);
+		free(risaf);
+		return TEE_SUCCESS;
+	}
+
+	strncpy(risaf->pdata.risaf_name, fdt_get_name(fdt, node, NULL),
```

Comment:
If not checking `strncpy()` return value, could you add an inline comment above like 
`/* Silently allow truncated names */`.